service_objects/fields.py

```diff
@@ -58,3 +59,80 @@ def clean(self, values):
             item_forms.append(item_form)
 
         return item_forms
+
+
+class ModelField(forms.Field):
+    """
+    A field for :class:`Service` that accepts an object of the specified
+    :class:`Model`.
+
+        :param model_class: Django :class:`Model` or dotted string of :
+            class:`Model` name
+        :param allow_unsaved: Whether the object is required to be saved to
+            the database
+    """
+    error_model_class = _("%(cls_name)s(%(model_class)r) is invalid.  First "
+                          "parameter of ModelField must be either a model or a "
+                          "model name.")
+    error_type = _("Objects needs to be of type %(model_class)r")
+    error_unsaved = _("Unsaved objects are not allowed.")
+
+    def __init__(self, model_class, allow_unsaved=False, *args, **kwargs):
+        super(ModelField, self).__init__(*args, **kwargs)
+
+        try:
+            model_class._meta.model_name
+        except AttributeError:
+            assert isinstance(model_class, str), self.error_model_class % {
+                'cls_name': self.__class__.__name__,
+                'model_class': model_class
+            }
+
+        self.model_class = model_class
+        if isinstance(model_class, str):
+            label = model_class.split('.')
+            app_label = ".".join(label[:-1])
+            model_name = label[-1]
+            self.model_class = apps.get_model(app_label, model_name)
+
+        self.allow_unsaved = allow_unsaved
+
+    def clean(self, value):
+        self.check_type(value)
+        self.check_unsaved(value)
+
+        return value
+
+    def check_type(self, item):
+        if not isinstance(item, self.model_class):
+            raise ValidationError(self.error_type % {
+                'model_class': self.model_class
+                }
+            )
+
+    def check_unsaved(self, item):
+        if (self.allow_unsaved is False and item.id is None):
+            raise ValidationError(self.error_unsaved)
+
+
+class MultipleModelField(ModelField):
+    """
+    A multiple model version of :class:`ModelField`, will check each passed
+    in object to match the specified :class:`Model`.
+
+    Will always return data as a list even if only a single object is passed in.
+
+        :param model_class: Django :class:`Model` or dotted string of :
+            class:`Model` name
+        :param allow_unsaved: Whether the object is required to be saved to
+            the database
+
+    """
+    def clean(self, values):
+        if not isinstance(values, list):
```

Comment:
I'm not sure if it's best for MultipleModelField to accept both single instances and a list of instances. Maybe it shouldn't accept a single instance? I'm happy to just let it raise 'is not iterable' when it reaches the for loop.